### PR TITLE
feat(debug): deduplicate response headers

### PR DESCRIPTION
**Pull request by tzssangglass**
### What this PR does / why we need it:

fix: #4611

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by tzssangglass** [comment]
@spacewander pls pre-review it, is this an available way?

**Comment by tzssangglass** [comment]
The response header will now have two `Apisix-Plugins`, one bound to the global and one bound to the non-global.

File: apisix/plugin.lua [hunk 1]
```diff
@@ -299,18 +299,33 @@ local function trace_plugins_info_for_debug(plugins)
...
-        core.response.add_header("Apisix-Plugins", core.table.concat(t, ", "))
+        if ctx then
+            local debug_headers = ctx.debug_headers
+            if debug_headers then
+                if type(t) == "string" then
+                    core.table.insert(debug_headers, t)
+                end
+                if type(t) == "table" then
+                    for i, v in ipairs(t) do
+                        core.table.insert(debug_headers, v)
```
Comment (spacewander) [inline apisix/plugin.lua:310]: Why not use a hash?

File: apisix/plugin.lua [hunk 2]
```diff
@@ -299,18 +299,33 @@ local function trace_plugins_info_for_debug(plugins)
         core.table.insert(t, plugins[i].name)
     end
     if is_http and not ngx.headers_sent then
-        core.response.add_header("Apisix-Plugins", core.table.concat(t, ", "))
+        if ctx then
+            local debug_headers = ctx.debug_headers
+            if debug_headers then
+                if type(t) == "string" then
```
Comment (spacewander) [inline apisix/plugin.lua:305]: t is always a table

File: t/debug/debug-mode.t [hunk 3]
```diff
@@ -238,11 +238,9 @@ passed
 GET /hello
 --- yaml_config eval: $::yaml_config
 --- response_headers
-Apisix-Plugins: response-rewrite, limit-conn, limit-count, response-rewrite, response-rewrite
+Apisix-Plugins: response-rewrite, limit-conn, limit-count
 --- response_body
 yes
---- error_log
```
Comment (spacewander) [inline t/debug/debug-mode.t:244]: We still need to check the `core.log.warn("Apisix-Plugins: ", core.table.concat(t, ", "))`

File: apisix/plugin.lua (hunk 2 above)
Comment (tzssangglass) [reply apisix/plugin.lua]: done

File: t/debug/debug-mode.t (hunk 3 above)
Comment (tzssangglass) [reply t/debug/debug-mode.t:244]: done

File: apisix/plugin.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/plugin.lua]: done

**Review by spacewander — APPROVED** [review]

**Review by shuaijinchao — APPROVED** [review]

**Pull request merged**